scripts/check_frontend_test_coverage.py

```diff
@@ -300,6 +299,7 @@
     'social-buttons.component.ts',
     'solution-editor.directive.ts',
     'solution-explanation-editor.directive.ts',
+    'state-classifier-mapping.service.ts',
```

Comment:
Can you check with @nithusha21 and @U8NWXD about this? We try not to introduce new files here (see the note at the top of the file).

Reply:
I see. I couldn't add tests for `StateClassifierMappingService.initializeClassifierDataForState` because I wasn't able to mock the `ClassifierDataBackendApiService` which is used by this function. I will talk with them for this.

Reply:
Ah ok, @srijanreddy98 might actually be the person you want to talk to here as well I think.

Reply:
Create a variable in describe block:
```
let classifierDataBackendApiService: ClassifierDataBackendApiService;
```
Inside the beforeEach block that contains TestBed.configureTestingModule, add:
`classifierDataBackendApiService = TestBed.inject(ClassifierDataBackendApiService);`

Inside an it block, do the following to mack:
```
spyOn(classifierDataBackendApiService, 'getClassifierData').and.callFake(() => { 
  return new Promise((resolve) => {
        resolve(DummyData you want to resolve);
  });
});
```
I have noticed that you have kept try-catch as well. So maybe try to throw an error in the callFake and that will help you cover the catch part. Let me know if you need help with anything else.

Reply:
That is also not working. The fake is being called (checked through logs) but the promise on the receiver side doesn't get resolved successfully.

Reply:
Done. Thanks @srijanreddy98 for getting this done.